pkg/network/go/bininspect/utils.go

```diff
@@ -0,0 +1,126 @@
+package bininspect
+
+import (
+	"debug/dwarf"
+	"debug/elf"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/network/go/asmscan"
+	"github.com/DataDog/datadog-agent/pkg/network/go/binversion"
+	"github.com/go-delve/delve/pkg/goversion"
+)
+
+// GetArchitecture returns the `runtime.GOARCH`-compatible names of the architecture.
+// Only returns a value for supported architectures.
+func GetArchitecture(elfFile *elf.File) (GoArch, error) {
+	switch elfFile.FileHeader.Machine {
+	case elf.EM_X86_64:
+		return GoArchX86_64, nil
+	case elf.EM_AARCH64:
+		return GoArchARM64, nil
+	}
+
+	return "", fmt.Errorf("unsupported architecture")
+}
+
+// HasDwarfInfo attempts to parse the DWARF data and look for any records.
+// If it cannot be parsed or if there are no DWARF info records,
+// then it assumes that the binary has been stripped.
+func HasDwarfInfo(elfFile *elf.File) (*dwarf.Data, bool) {
+	dwarfData, err := elfFile.DWARF()
+	if err != nil {
+		return nil, false
+	}
+
+	infoReader := dwarfData.Reader()
+	if firstEntry, err := infoReader.Next(); err == nil && firstEntry != nil {
+		return dwarfData, true
+	}
+
+	return nil, false
+}
+
+// GetAllSymbolsByName returns all the elf file's symbols mapped by their name.
+func GetAllSymbolsByName(elfFile *elf.File) (map[string]elf.Symbol, error) {
+	regularSymbols, regularSymbolsErr := elfFile.Symbols()
+	dynamicSymbols, dynamicSymbolsErr := elfFile.DynamicSymbols()
+
+	// Only if we failed getting both regular and dynamic symbols - then we abort.
```

Comment:
fixed!

Reply:
I meant - why abort only if both got error? we have some logic afterwards that handles both of them so it's seems weird

Reply:
Cause we might be able to run properly even if just one of them is ok

Reply:
Hmm, I don't know exactly what can be the error in this specific case, but I personally don't like error to be ignored as it can possibly affect the logic afterwards. Maybe log at least? not sure if it will be helpful though I think only some tests could verify this. Anyway if you feel confident with it feel free to resolve this comment :)